Source/DafnyLanguageServer/Workspace/GutterIconAndHoverVerificationDetailsManager.cs

```diff
@@ -309,13 +309,40 @@ public class GutterIconAndHoverVerificationDetailsManager : IGutterIconAndHoverV
           targetMethodNode.StatusVerification = finalOutcome;
         }
 
+        var options = compilation.Program.Reporter.Options;
+        ReportVacuityAndRedundantAssumptionsChecks(compilation, implementation, verificationResult, options);
+
         targetMethodNode.PropagateChildrenErrorsUp();
         targetMethodNode.RecomputeAssertionBatchNodeDiagnostics();
         PublishGutterIcons(compilation, uri, true);
       }
     }
   }
 
+  private static void ReportVacuityAndRedundantAssumptionsChecks(CompilationAfterResolution compilation,
+    Implementation implementation, VerificationResult verificationResult, DafnyOptions options) {
+    if (!options.Get(CommonOptionBag.WarnContradictoryAssumptions)
+        && !options.Get(CommonOptionBag.WarnRedundantAssumptions)
+       ) {
+      return;
+    }
+
+    var vcResults = verificationResult.VCResults.Select(r =>
+      new DafnyConsolePrinter.VCResultLogEntry(r.vcNum, r.startTime, r.runTime, r.outcome,
```

Comment:
I see duplication between line 330-337 here and `DafnyConsolePrinter.ReportEndVerifyImplementation`. Could you extract a reusable method from `DafnyConsolePrinter.ReportEndVerifyImplementation` to remove the duplication?